flashlight/fl/examples/RnnClassification.cpp

```diff
@@ -0,0 +1,260 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+/* Approximate re implementation of the char rnn PyTorch tutorial:
+   https://pytorch.org/tutorials/intermediate/char_rnn_classification_tutorial.html
+   Dataset: wget https://download.pytorch.org/tutorial/data.zip
+*/
+
+#include <iomanip>
+#include <iostream>
+// Check with reviewers: what about enabling C++17 at least for the examples and get access to (at least) std::filesystem ?
+//#include <filesystem>
+#include <stdexcept>
+#include <sstream>
+#include <ostream>
+#include <unordered_map>
+#include <map>
+#include <vector>
+#include <random>
+
+#include "flashlight/fl/flashlight.h"
+
+using namespace fl;
+
+// return a random int between [mini, maxi]
+int randi(int mini, int maxi) {
+  if(maxi<mini) 
+    std::swap(maxi,mini);
+  return rand()%(maxi-mini + 1) + mini; 
+}
+
+class ClassDataset : public Dataset {
+ public:
+  using names=std::vector<std::string>;
+  std::map<std::string, names> datasets;
+  static std::map<std::string, unsigned> label2ID;
+  static std::map<unsigned, std::string> ID2label;
+  unsigned totalnumexamples=0;
+
+  // read the folder/lang.txt file and register the examples in the datasets map
+  void read(const std::string folder, const std::string lang) {
+    FL_VLOG(fl::INFO) << "Opening " << folder << "/" << lang << ".txt ...";
+    std::ifstream file(folder + "/" + lang +".txt");
```

Comment:
you could use here https://github.com/facebookresearch/flashlight/blob/master/flashlight/lib/common/System.cpp#L31 to get the full path

Reply:
done